<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>623</width>
    <height>434</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Create Character</string>
  </property>
  <property name="sizeGripEnabled" stdset="0">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>587</width>
        <height>611</height>
       </rect>
      </property>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Name:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="nameLineEdit"/>
         </item>
         <item>
          <widget class="QPushButton" name="randomNamePushButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Randomize</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Race:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="SelectRaceComboBox" name="selectRaceComboBox"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>Gender:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QRadioButton" name="maleRadioButton">
           <property name="text">
            <string>Male</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">genderButtonGroup</string>
           </attribute>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="femaleRadioButton">
           <property name="text">
            <string>Female</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">genderButtonGroup</string>
           </attribute>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Attributes:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="5" column="2">
            <widget class="QPushButton" name="rerollStrengthPushButton_6">
             <property name="text">
              <string>reroll</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="rerollStrengthPushButton_2">
             <property name="text">
              <string>reroll</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="dexterityValueLabel">
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="constitutionValueLabel">
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Strength:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="strengthValueLabel">
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QLabel" name="charismaValueLabel">
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="4" column="2">
            <widget class="QPushButton" name="rerollStrengthPushButton_5">
             <property name="text">
              <string>reroll</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <widget class="QPushButton" name="rerollStrengthPushButton">
             <property name="text">
              <string>reroll</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Intelligence:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Wisdom:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="rerollStrengthPushButton_3">
             <property name="text">
              <string>reroll</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QLabel" name="wisdomValueLabel">
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="3" column="2">
            <widget class="QPushButton" name="rerollStrengthPushButton_4">
             <property name="text">
              <string>reroll</string>
             </property>
            </widget>
           </item>
           <item row="6" column="2">
            <widget class="QPushButton" name="rerollAllPushButton_8">
             <property name="text">
              <string>reroll all</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QLabel" name="intelligenceValueLabel">
             <property name="text">
              <string>TextLabel</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Dexterity:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Constitution:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Charisma:</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>Skills:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QTreeView" name="skillTreeView"/>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>Personality:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QTreeView" name="personalityTreeView"/>
       </item>
       <item row="6" column="0">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>History:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
         <property name="labelClass" stdset="0">
          <string>fieldName</string>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QTextEdit" name="historyTextEdit"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SelectRaceComboBox</class>
   <extends>QComboBox</extends>
   <header>jgrpg/SelectRaceComboBox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Widget</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Widget</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>301</x>
     <y>427</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>87</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollStrengthPushButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollStrength()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>118</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>74</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollStrengthPushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollDexterity()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>184</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>102</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollStrengthPushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollConstitution()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>131</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollStrengthPushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollIntelligence()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>259</y>
    </hint>
    <hint type="destinationlabel">
     <x>395</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollStrengthPushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollWisdom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>296</y>
    </hint>
    <hint type="destinationlabel">
     <x>396</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollStrengthPushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollCharisma()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>333</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>217</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rerollAllPushButton_8</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>rerollAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>370</y>
    </hint>
    <hint type="destinationlabel">
     <x>390</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>randomNamePushButton</sender>
   <signal>clicked()</signal>
   <receiver>Widget</receiver>
   <slot>randomizeName()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>533</x>
     <y>31</y>
    </hint>
    <hint type="destinationlabel">
     <x>564</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>accept()</slot>
  <slot>reject()</slot>
  <slot>rerollStrength()</slot>
  <slot>rerollDexterity()</slot>
  <slot>rerollConstitution()</slot>
  <slot>rerollIntelligence()</slot>
  <slot>rerollWisdom()</slot>
  <slot>rerollCharisma()</slot>
  <slot>rerollAll()</slot>
  <slot>randomizeName()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="genderButtonGroup"/>
 </buttongroups>
</ui>
